src/index.ts

```diff
@@ -2,12 +2,26 @@ import { rpcErrors } from '@metamask/rpc-errors';
 import type { OnRpcRequestHandler } from '@metamask/snaps-sdk';
 import { z } from 'zod';
 
-import { getPublicEntropyKey, signMessageWithEntropyKey } from './entropy-keys';
+import {
+  decryptMessage,
+  getEncryptionPublicKey,
+  getPublicEntropyKey,
+  signMessageWithEntropyKey,
+} from './entropy-keys';
 
 const SignMessageParamsSchema = z.object({
   message: z.string().startsWith('metamask:'),
 });
 
+const DecryptMessageParamsSchema = z.object({
+  data: z.object({
+    version: z.string().regex(/^x25519-xsalsa20-poly1305$/u),
```

Comment:
Can we replace the regex with an exact equals string?

Reply:
I wasn't able to find an `equals` constraint for strings in the Zod API: https://zod.dev/?id=strings
Happy to take suggestions :)

Reply:
Oooh, interesting. I'll have a look!

Reply:
Try z.literal: https://zod.dev/?id=literals

Once we need to support multiple version, we can move to z.union or z.enum.

```suggestion
    version: z.literal('x25519-xsalsa20-poly1305'),
```